### PR TITLE
🌱 RuntimeSDK/client: use random port for unit test server

### PR DESCRIPTION
**Pull request by sbueringer**
**What this PR does / why we need it**:
I noticed that when running the tests locally I hit the issue that parallel running tests fail because the port is already used.

I assume Go only starts more tests (in this case of the same package) in parallel if the CPU core/thread count is high enough :) 

But of course this also fails if something else is using the port locally which can easily happen with 9090.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: internal/runtime/client/client_test.go [hunk 1]
```diff
@@ -1150,10 +1153,7 @@ func response(status runtimehooksv1.ResponseStatus) testServerResponse {
 	}
 }
 
-func createSecureTestServer(g *WithT, server testServerConfig) *httptest.Server {
-	l, err := net.Listen("tcp", server.hostPort)
-	g.Expect(err).NotTo(HaveOccurred())
-
+func createSecureTestServer(server testServerConfig) *httptest.Server { // FIXME
```
Comment (killianmuldoon) [inline internal/runtime/client/client_test.go:1156]: What's the FIXME for?

Comment (sbueringer) [reply internal/runtime/client/client_test.go:1156]: Dropping the `g` parameter which I did :)

File: internal/runtime/client/client_test.go [hunk 2]
```diff
@@ -530,7 +528,8 @@ func TestClient_CallExtension(t *testing.T) {
 	validExtensionHandlerWithFailPolicy := runtimev1.ExtensionConfig{
 		Spec: runtimev1.ExtensionConfigSpec{
 			ClientConfig: runtimev1.ClientConfig{
-				URL:      pointer.String(fmt.Sprintf("https://%s/", testHostPort)),
+				// Set a dummy URL, in test cases where we start the test server the URL will be overridden.
```
Comment (vincepri) [inline internal/runtime/client/client_test.go:531]:
```suggestion
				// Set a fake URL, in test cases where we start the test server the URL will be overridden.
```

File: internal/runtime/client/client_test.go [hunk 3]
```diff
@@ -552,7 +551,8 @@ func TestClient_CallExtension(t *testing.T) {
 	validExtensionHandlerWithIgnorePolicy := runtimev1.ExtensionConfig{
 		Spec: runtimev1.ExtensionConfigSpec{
 			ClientConfig: runtimev1.ClientConfig{
-				URL:      pointer.String(fmt.Sprintf("https://%s/", testHostPort)),
+				// Set a dummy URL, in test cases where we start the test server the URL will be overridden.
```
Comment (vincepri) [inline internal/runtime/client/client_test.go:554]:
```suggestion
				// Set a fake URL, in test cases where we start the test server the URL will be overridden.
```

File: internal/runtime/client/client_test.go [hunk 4]
```diff
@@ -780,7 +781,8 @@ func TestClient_CallAllExtensions(t *testing.T) {
 	extensionConfig := runtimev1.ExtensionConfig{
 		Spec: runtimev1.ExtensionConfigSpec{
 			ClientConfig: runtimev1.ClientConfig{
-				URL:      pointer.String(fmt.Sprintf("https://%s/", testHostPort)),
+				// Set a dummy URL, in test cases where we start the test server the URL will be overridden.
```
Comment (vincepri) [inline internal/runtime/client/client_test.go:784]:
```suggestion
				// Set a fake URL, in test cases where we start the test server the URL will be overridden.
```

**Comment by sbueringer** [comment]
@vincepri Done, PTAL

**Review by killianmuldoon — COMMENTED** [review]
/lgtm

**Review by vincepri — APPROVED** [review]
/approve

**Pull request merged**